### PR TITLE
lab2

### PR DESCRIPTION
**Pull request by EvgeniyVashinko**
lab2

### CODE REVIEW

File: lab2/SuperTimer/app/src/main/java/com/example/supertimer/data/TimerDao.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+@Dao
+public interface TimerDao {
+    @Query("SELECT * FROM timer")
+    List<Timer> getAllTimers();
+
+    @Query("SELECT * FROM timer")
+    LiveData<List<Timer>> getAllTimersLiveData();
+
+    @Query("SELECT * FROM timer WHERE id LIKE :timerId LIMIT 1")
```
Comment (gmltA) [inline lab2/SuperTimer/app/src/main/java/com/example/supertimer/data/TimerDao.java:23]: Why do use `LIKE` here?

File: lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        size = Float.parseFloat(sp.getString("fontSize", "1.0"));
+        lang = sp.getString("language", "RU");
+        nMode = sp.getBoolean("nightMode",  false);
+
+        if (nMode) {
+            setTheme(R.style.my_theme_dark);
+        } else {
+            setTheme(R.style.my_theme_light);
+        }
```
Comment (gmltA) [inline lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/MainActivity.java]: Can you extract identical customisation code into abstract Activity class and inherit it everywhere you need it?

File: lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/TimerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    private Context context;
+    private int size;
+
+    public TimerAdapter(Context context, int resource, List<Timer> timers) {
+        super(context, resource, timers);
+        this.context = context;
+        this.timers = timers;
+        this.layout = resource;
+        this.inflater = LayoutInflater.from(context);
+        this.size = 12;
```
Comment (gmltA) [inline lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/TimerAdapter.java]: This thing doesn't look good either.

**Review by gmltA — CHANGES_REQUESTED** [review]
Not bad. But I want to see some minor improvements first.

File: lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/MainActivity.java (hunk 2 above)
Comment (EvgeniyVashinko) [reply lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/MainActivity.java]: sounds good

File: lab2/SuperTimer/app/src/main/java/com/example/supertimer/data/TimerDao.java (hunk 1 above)
Comment (EvgeniyVashinko) [reply lab2/SuperTimer/app/src/main/java/com/example/supertimer/data/TimerDao.java:23]: when i read documentation for room, i saw a class-example there. i copied and fixed it for my application to save time. 
in this case we can replace LIKE with '='

File: lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/TimerAdapter.java (hunk 3 above)
Comment (EvgeniyVashinko) [reply lab2/SuperTimer/app/src/main/java/com/example/supertimer/view/main/TimerAdapter.java]: i fixed it

**Comment by EvgeniyVashinko** [comment]
I passed 1st and 3rd labs. Is it all?

**Review by gmltA — APPROVED** [review]
That's it. Good luck!

**Comment by EvgeniyVashinko** [comment]
Thanks, you too

**Pull request merged**